File: BC-W3/Java/java/java/Pen&Paper/Paper.java
public class Paper {
    private StringBuffer content;
    private int maxSymbols;
    private int symbols;
    
    public Paper(int maxSymbols) {
        this.symbols = 0;
        this.maxSymbols = maxSymbols;
        content = new StringBuffer();
    }

    public Paper() {
        this(4096); 
    }

    public int getMaxSymbols() {
        return maxSymbols;
    }
    
    public int getSymbols() {
        return symbols;
    }
    
    public void addContent(String message) throws OutOfSpaceException {
        if ( symbols == maxSymbols ) {
            throw new OutOfSpaceException();
        }
        
        this.symbols += message.length();
        
        if ( this.symbols > this.maxSymbols ) {
            int temp = message.length() - (this.symbols - this.maxSymbols);
            
            this.symbols = this.maxSymbols;
            content.append(message.substring(0, temp));
        } else {
            content.append(message);
        }
    }
    public void show() {
        System.out.println(content);
    }
}

File: BC-W3/TaskCpp/bcw4/Pen&Paper/Paper.cpp
#include "Paper.h"

Paper::Paper(int maxSymbols) {
    this->maxSymbols = maxSymbols;
    this->symbols = 0;
}
Paper::~Paper() {}

int Paper::getMaxSymbols() const {
    return this->maxSymbols;
}
int Paper::getSymbols() const {
    return this->symbols;
}

void Paper::addContent(const std::string& message) {
    if ( this->symbols == this->maxSymbols ) {
        throw OutOfSpace();
    }
    this->symbols += message.size();
    
    if ( this->symbols > this->maxSymbols ) {
        int temp = message.size() - (this->symbols - this->maxSymbols);
        
        this->symbols = this->maxSymbols;
        this->content += message.substr(0, temp);
    } else {
        this->content += message; 
    }
}
void Paper::show() const {
    std::cout << this->content << std::endl;
}

